tests/unit/core/api/test_abuse.js

```diff
@@ -0,0 +1,85 @@
+import * as api from 'core/api';
+import { reportAddon } from 'core/api/abuse';
+import {
+  dispatchClientMetadata,
+  dispatchSignInActions,
+  fakeAddon,
+} from 'tests/unit/amo/helpers';
+import {
+  createApiResponse,
+  createFakeAddonAbuseReport,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let mockApi;
+
+  beforeEach(() => {
+    mockApi = sinon.mock(api);
+  });
+
+  describe('reportAddon', () => {
+    function mockResponse({ addon, message }) {
+      return createApiResponse({
+        jsonData: createFakeAddonAbuseReport({ addon, message }),
+      });
+    }
+
+    it('should allow anonymous users to report an add-on', () => {
+      const apiState = dispatchClientMetadata().store.getState().api;
+      const message = 'I do not like this!';
+
+      mockApi
+        .expects('callApi')
+        .withArgs({
+          auth: true,
+          endpoint: 'abuse/report/addon',
+          method: 'POST',
+          params: { addon: 'cool-addon', message },
+          state: apiState,
+        })
+        .once()
+        .returns(mockResponse({
+          addon: { ...fakeAddon, slug: 'cool-addon' },
+          message,
+        }));
+      return reportAddon({
+        addonSlug: 'cool-addon',
+        api: apiState,
+        message,
+      })
+        .then(() => {
+          mockApi.verify();
+        });
+    });
+
+    it('should allow signed-in users to report an add-on', () => {
```

Comment:
It would be nice to update this spec to avoid confusion. The API call applies to both signed in or anonymous users.

Reply:
Yeah I actually think now that `auth: true` is always set this can just be removed.